### PR TITLE
Skip dispatch for `is_floating_point()`

### PR DESCRIPTION
**Pull request by iramazanli**

### CODE REVIEW

**Comment by ngimel** [comment]
Why don't you use manual_cpp_bindings to skip dispatch? #49092

**Comment by anjali411** [comment]
@ngimel is right; @iramazanli you could follow this PR that skips dispatch for `is_complex()` https://github.com/pytorch/pytorch/pull/50054

File: aten/src/ATen/native/LinearAlgebra.cpp [hunk 1]
```diff
@@ -781,7 +781,7 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
   } else if (at::hasMKL() && ((
             self_or_result.scalar_type() != kHalf &&
             self_or_result.scalar_type() != kBFloat16 &&
-            at::native::is_floating_point(self_or_result)) ||
+            at::is_floating_point(self_or_result)) ||
```
Comment (anjali411) [inline aten/src/ATen/native/LinearAlgebra.cpp:784]: you don't need this change anymore

File: aten/src/ATen/templates/TensorMethods.cpp [hunk 2]
```diff
@@ -153,7 +153,6 @@ bool Tensor::is_metal() const {
   return impl_->is_metal();
 }
 
-
```
Comment (anjali411) [inline aten/src/ATen/templates/TensorMethods.cpp:156]: please undo this change

File: tools/autograd/templates/python_torch_functions.cpp [hunk 3]
```diff
@@ -376,6 +376,14 @@ static PyObject * THPVariable_as_tensor(PyObject* self, PyObject* args, PyObject
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject * THPVariable_is_floating_point(PyObject* self)
```
Comment (anjali411) [inline tools/autograd/templates/python_torch_functions.cpp:379]: you don't need this anymore

File: tools/autograd/templates/python_variable_methods.cpp [hunk 4]
```diff
@@ -178,6 +178,18 @@ static PyObject * THPVariable_has_names(PyObject* self_, PyObject* args)
   END_HANDLE_TH_ERRORS
 }
 
+// implemented on the python object to avoid dispatch overhead
+static PyObject * THPVariable_is_floating_point(PyObject* self_, PyObject* args)
```
Comment (anjali411) [inline tools/autograd/templates/python_variable_methods.cpp:182]: you don't need this change anymore

File: tools/pyi/gen_pyi.py [hunk 5]
```diff
@@ -451,6 +451,7 @@ def gen_pyi(native_yaml_path: str, deprecated_yaml_path: str, fm: FileManager) -
         'is_meta': ['is_meta: _bool'],
         'is_mkldnn': ['is_mkldnn: _bool'],
         'is_vulkan': ['is_vulkan: _bool'],
+        'is_floating_point': ['is_floating_point: _bool'],
```
Comment (anjali411) [inline tools/pyi/gen_pyi.py:454]: you don't need this anymore

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 6]
```diff
@@ -542,6 +542,16 @@ RegisterOperators reg(
            pack(stack, result);
          },
          aliasAnalysisFromSchema()),
+     OperatorGenerator(
```
Comment (anjali411) [inline torch/csrc/jit/runtime/register_prim_ops.cpp:545]: you don't need this anymore

**Comment by anjali411** [comment]
left some comments. long story short, you don't need a lot of changes that you have anymore since you have set the `manual_cpp_bindings: True` which takes a care of the code you have rn. Also, I think you need to rebase this PR on the latest master.

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 7]
```diff
@@ -542,8 +542,6 @@ RegisterOperators reg(
            pack(stack, result);
          },
          aliasAnalysisFromSchema()),
-     // these ops are generic over the list element type.
```
Comment (anjali411) [inline torch/csrc/jit/runtime/register_prim_ops.cpp:545]: please undo this change

File: aten/src/ATen/templates/Functions.h [hunk 8]
```diff
@@ -146,4 +146,7 @@ inline bool is_complex(const Tensor& tensor) {
   return tensor.is_complex();
 }
 
+inline bool is_floating_type(const Tensor& tensor) {
```
Comment (anjali411) [inline aten/src/ATen/templates/Functions.h:149]: should be `inline bool is_floating_point(const Tensor& tensor) {`

Also I'd suggest locally compiling the changes before pushing them :)
